### DISCUSSION TITLE
Allow writing of an empty string as an attribute?

### DISCUSSION THREAD

**Discussion by gtokic** [geospace-code/h5fortran#44] [Q&A] [answered]
It seems that the latest release v4.10.2 no longer allows writing an empty string as an attribute (v4.2.11 allowed it). 

Of course users could check for `len(string)` before trying to write `string` as an attribute
```fortran
call h5f_p%writeattr(key,'attr_name',string)
```
but it seems to me that it is useful for h5fortran to handle that natively, as it did before. 

I think inserting a single line after line 22 in `attr_write.inc` (v4.10.2) would do the trick.
```fortran
if (charlen == 0) charlen = 1
```

**Comment by scivision** [comment, marked as answer]
Thanks, this was fixed in https://github.com/geospace-code/h5fortran/commit/07c7d62268ef683456e3eb82276bfb879ce3f8ea
